server/src/main.rs

```diff
@@ -147,31 +146,34 @@ fn main() {
         std::process::exit(0);
     }
 
-    let old_config = match Configuration::load(config_path.as_deref(), cli_args.opts_overrides) {
-        Ok(c) => c,
-        Err(e) => {
-            // We cannot use tracing here as it's not configured yet
-            eprintln!("{}", e.decorate());
-            eprintln!("{:#?}", RestateCode::from(&e));
-            std::process::exit(EXIT_CODE_FAILURE);
-        }
-    };
+    let old_config =
+        match restate_node::Configuration::load(config_path.as_deref(), cli_args.opts_overrides) {
+            Ok(c) => c,
+            Err(e) => {
+                // We cannot use tracing here as it's not configured yet
+                eprintln!("{}", e.decorate());
+                eprintln!("{:#?}", RestateCode::from(&e));
+                std::process::exit(EXIT_CODE_FAILURE);
+            }
+        };
 
     restate_node::set_current_config(old_config);
     restate_types::config::set_current_config(config);
 
+    let old_config = restate_node::Configuration::pinned();
```

Comment:
By pinning the configuration aren't we blocking one slot of ArcSwap permanently because we hold on to the `old_config` until the very end of the application?